### DISCUSSION TITLE
Copy text from Windows Terminal buffer to clipboard

### DISCUSSION THREAD

**Discussion by pcgeek86** [microsoft/terminal#13754] [General]
I would love to be able to copy text to the Windows clipboard, from the Windows Terminal buffer, so I can use it in a command. For example, see #6632 which discusses iTerm2-like auto-completion.

💡 Where would I start with the Windows Terminal APIs, to be able to create such an application?

Imagine that you run this command, and receive the following output in the Windows Terminal buffer:

```
C:\Users\TrevorSullivan> kubectl get node
NAME                           STATUS   ROLES    AGE    VERSION
lke68670-106518-62f56ccf7428   Ready    <none>   4d9h   v1.23.6
lke68670-106518-62f56ccf8947   Ready    <none>   4d9h   v1.23.6
C:\Users\TrevorSullivan>
```

Let's say that I want to copy `lke68670-106518-62f56ccf7428` to my clipboard, and insert it into the Windows Terminal shell prompt.
I would envision creating a utility that could at the very least:

- Query the Windows Terminal buffer
- Run a regular expression against the buffer text
- Copy the first regex match to the clipboard

Imagine that I create a utility called `wtcopy` that accomplished the above actions. 
Using the following command, I could accomplish the above tasks.

```
C:\Users\TrevorSullivan> wtcopy 7428
```

**Comment by zadjii-msft** [comment]
You might be interested in #8274.

There's not really "terminal APIs" per-say. But would something like <kbd>ctrl+shift+f</kbd>, {type whatever you're looking for}, <kbd>tab</kbd>, <kbd>ctrl+shift+c</kbd> work for this scenario?

**Reply by pcgeek86** [reply]
I'd like to reduce the amount of typing by using a regular expression search, but it appears that Windows Terminal search does not support regex, like VSCode search/replace does. The iTerm2-like auto-completion is really important, because it is "intelligent" and able to find "words" and other custom entities. If you're not familiar with iTerm2's auto-completion feature, I would strongly encourage you to check it out; it's really slick, and I can't wait to see that in Windows Terminal. 👍🏻

**Reply by snowman** [reply]
There needs the terminal APIs indeed, so I can use launcher app to provide context-menu for example around the cursor

**Reply by snowman** [reply]
With few APIs exposed, the compositions are endless, which creates a way to communicate with other tools
